ha/setup/ha_setup.py

```diff
@@ -0,0 +1,343 @@
+
+class PostInstallCmd(Cmd):
+    """
+    PostInstall Setup Cmd
+    """
+    name = "post_install"
+
+    def __init__(self, args: dict):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process post_install command.
+        """
+
+        Log.info("Processing post_install command")
+        # Create a directory and copy config file
+        if os.path.exists(const.HA_CONFIG_FILE):
+            os.remove(const.HA_CONFIG_FILE)
+        os.makedirs(const.CONFIG_DIR, exist_ok=True)
+        shutil.copyfile(const.SOURCE_CONFIG_FILE, const.HA_CONFIG_FILE)
+        Log.info(f"{self.name}: Copied HA config file.")
+        # Pre-requisite checks are done here.
+        # Make sure the pacemaker, corosync and pcs packages have been installed
+        PkgV().validate('rpms', const.PCS_CLUSTER_PACKAGES)
+        Log.info("post_install command is successful")
+
+class ConfigCmd(Cmd):
+    """
+    Setup Config Cmd
+    """
+    name = "config"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process config command.
+        """
+        Log.info("Processing config command")
+        # Read machine-id and using machine-id read minion name from confstore
+        # This minion name will be used for adding the node to the cluster.
+        nodelist = []
+        command = "cat /etc/machine-id"
+        machine_id, err, rc = self._execute.run_cmd(command, check_error=True)
+        Log.info(f"Read machine-id. Output: {machine_id}, Err: {err}, RC: {rc}")
+        minion_name = Conf.get(self._index, f"cluster.server_nodes.{machine_id.strip()}")
+        nodelist.append(minion_name)
+        # The config step will be called from primary node alwasys,
+        # see how to get and use the node name then.
+
+        # Read cluster name and cluster user
+        cluster_name = Conf.get(self._index, 'corosync-pacemaker.cluster_name')
+        cluster_user = Conf.get(self._index, 'corosync-pacemaker.user')
+
+        # Read cluster user password and decrypt the same
+        cluster_id = Conf.get(self._index, 'cluster.cluster_id')
+        cluster_secret = Conf.get(self._index, 'corosync-pacemaker.secret')
+        key = Cipher.generate_key(cluster_id, 'corosync-pacemaker')
+        cluster_secret = Cipher.decrypt(key, cluster_secret.encode('ascii')).decode()
+
+        # Check if the cluster exists already, if yes skip creating the cluster.
+        output, err, rc = self._execute.run_cmd(const.PCS_CLUSTER_STATUS, check_error=False)
+        Log.info(f"Cluster status. Output: {output}, Err: {err}, RC: {rc}")
+        if rc != 0:
+            if(err.find("No such file or directory: 'pcs'") != -1):
+                Log.error("Cluster config failed; pcs not installed")
+                raise Exception("Cluster config failed; pcs not installed")
```

Comment:
This is already checked in post_install. No harm in keeping the code but Exception should be HaConfigError in config phase.

Reply:
done